custom-elements/custom-elements-registry/define.html

```diff
@@ -56,7 +56,22 @@
     }, `If constructor is ${t[0]}, should throw a TypeError`);
   });
 
-  // 2. If name is not a valid custom element name,
+  // 2. If constructor is an interface object whose corresponding interface
+  // either is HTMLElement or has HTMLElement in its set of inherited interfaces,
+  // throw a TypeError and abort these steps.
+  [
+    [ 'HTMLElement', HTMLElement ],
+    [ 'HTMLButtonElement', HTMLButtonElement ],
+    [ 'HTMLImageElement', HTMLImageElement ],
+  ].forEach(t => {
```

Comment:
I'd suggest also adding `Image`, `Audio`, and `HTMLMediaElement` as they are tricky cases. In fact the spec currently does not disallow `Image` and `Audio`, but it should; I will send a follow-up PR to HTML to fix that.